nrf802154_sniffer/nrf802154_sniffer.py

```diff
@@ -412,7 +411,8 @@ def fifo_writer(self, fifo, queue):
                         fh.write(packet)
                         fh.flush()
                     except IOError:
-                        pass
```

Comment:
But what if we didn't recevie any frame a long time after sniffing was stopped? It seems that we detect stopped sniffing only when we receive a frame and write it to the Wireshark fifo.

Reply:
Indeed, this does cause the extcap to be closed right after the last packet arrives. Currently I have no idea how to address this problem any other way.

Reply:
I think that long time ago we had a thread that was polling if this fifo is closed in some short intervals (like 500 ms), but this polling didn't work on Windows, that's why we switched to detection of broken control_in fifo. We can somehow go back to polling approach and fix problem on windows or (what I prefer) create a dummy toolbar with a dummy option until we add real toolbar functionality.

Reply:
Or... we can create a toolbar with "Log" button :)

Reply:
The dummy log button appears to create the control-in FIFO. As a temporary workaround, I think this would be acceptable (though ugly) until the toolbar channel selection is added.
I have found out that the tshark is able to close the extcap even without the control-in fifo (perhaps by sending the SIGTERM or SIGINT?). This is not the case for the Wireshark.
I haven't tested that behaviour on tshark on Windows, but I suspect from reading `dumpcap.c` source file, that there are most likely differences.

Select/poll/epoll calls can't be used either because it's not portable.

@stig-bjorlykke do you know any more reliable method for detecting a closed FIFO?

Reply:
@e-rk Usually ```with open(fifo, 'wb', 0) as fh:``` will detect that the pipe is closed from Wireshark when the capture is stopped. This is the way it's detected in the sample [extcap_example.py](https://github.com/wireshark/wireshark/blob/master/doc/extcap_example.py#L354) from the Wireshark source code and in the BLE sniffer.

The toolbar support is optional in extcap so it should not be needed to check for exit in the control_in or control_out pipes.

I did a short test with a ```self.stop_sig_handler()``` in the end of ```fifo_writer``` but it didn't really work so it's probably something else blocking it from exit. I'll have a closer look at this later if no one else figures this out.

Reply:
Thank you for response. I think that this is the same as the solution I proposed at the beginning - detect if `write` throws an exception and close the extcap. The `with` keyword is basically a `try except finally` clause. It will wait until a packet arrives, attempt to write to a fifo closed by the reading end and throw an exception. As @hubertmis mentioned, this will not work if we don't receive a single frame afterwards and the process will remain running in the background.